channeldb/channel.go

```diff
@@ -2340,6 +2393,12 @@ func SerializeHtlcs(b io.Writer, htlcs ...HTLC) error {
 	}
 
 	for _, htlc := range htlcs {
+		// Populate TLV stream for any additional fields contained
+		// in the TLV.
+		if err := htlc.serializeExtraData(); err != nil {
```

Comment:
Won't this information already be stored in `ExtraData` if you just take the opaque bytes from the update add HTLC message? Since we code, but then keep the blob as is.